Orange/data/table.py

```diff
@@ -1535,7 +1535,8 @@ def has_missing_class(self):
     def get_nan_frequency_attribute(self):
         if self.X.size == 0:
             return 0
-        return np.isnan(self.X).sum() / self.X.size
+        vals = self.X if not sp.issparse(self.X) else self.X.data
+        return np.isnan(vals).sum() / np.prod(self.X.shape)
 
     def get_nan_frequency_class(self):
```

Comment:
I think you should do the same for `get_nan_frequency_class`. The problem there is much smaller - while `Domain` and `Table` support multiple classes, many widgets don't so we don't recommend that and almost never use it. Still; it's a trivial change, and it may hunt us or somebody else some time in the future.

Reply:
Upon copying the changes to `get_nan_frequency_class` I decided I could also just merge the two into a single function that handles both `X` and `Y` (and possibly `metas` or `W` in the future). I am however curious if they should remain separate for some reason?

Reply:
No, I see no reason for separate functions given that array/matrix can be an argument.

Reply:
If you'd make a new `get_nan_freqency` function that can handle any part, keep in mind that we can not delete the old functions immediately. We'll need to deprecate them first.